src/oss/langgraph/test.mdx

```diff
@@ -1,3 +1,320 @@
+
+## Prerequisites
+
+:::python
+First, make sure you have [`pytest`](https://docs.pytest.org/) installed:
+
+```bash
+$ pip install -U pytest
+```
+:::
+
+:::js
+First, make sure you have [`vitest`](https://vitest.dev/) installed:
+
+```bash
+$ npm install -D vitest
+```
+:::
+
+## Getting started
+
+Because many LangGraph agents depend on state, a useful pattern is to create your graph before each test where you use it, then compile it within tests with a new checkpointer instance.
+
+The below example shows how this works with a simple, linear graph that progress from `node1` to `node4`. Each node updates a single state key:
+
+:::python
+```python
+import pytest
+
+from typing_extensions import TypedDict
+from langgraph.graph import StateGraph, START, END
+from langgraph.checkpoint.memory import MemorySaver
+
+@pytest.fixture
+def uncompiled_graph():
+    class MyState(TypedDict):
+        my_key: str
+
+    graph = StateGraph(MyState)
+    graph.add_node("node1", lambda state: {"my_key": "hello from node1"})
+    graph.add_node("node2", lambda state: {"my_key": "hello from node2"})
+    graph.add_node("node3", lambda state: {"my_key": "hello from node3"})
+    graph.add_node("node4", lambda state: {"my_key": "hello from node4"})
+    graph.add_edge(START, "node1")
+    graph.add_edge("node1", "node2")
+    graph.add_edge("node2", "node3")
+    graph.add_edge("node3", "node4")
+    graph.add_edge("node4", END)
+    return graph
+
+def test_basic_agent_execution(uncompiled_graph):
+    checkpointer = MemorySaver()
+    compiled_graph = uncompiled_graph.compile(checkpointer=checkpointer)
+    result = compiled_graph.invoke(
+        {"my_key": "initial_value"},
+        config={"configurable": {"thread_id": "1"}}
+    )
+    assert result["my_key"] == "hello from node4"
+```
+:::
+
+:::js
+```ts
+import { test, expect } from 'vitest';
+import {
+  StateGraph,
+  START,
+  END,
+  MemorySaver,
+} from '@langchain/langgraph';
+import { z } from "zod/v4";
+
+const State = z.object({
+  my_key: z.string(),
+});
+
+const createGraph = () => {
+  return new StateGraph(State)
+    .addNode('node1', (state) => ({ my_key: 'hello from node1' }))
+    .addNode('node2', (state) => ({ my_key: 'hello from node2' }))
+    .addNode('node3', (state) => ({ my_key: 'hello from node3' }))
+    .addNode('node4', (state) => ({ my_key: 'hello from node4' }))
+    .addEdge(START, 'node1')
+    .addEdge('node1', 'node2')
+    .addEdge('node2', 'node3')
+    .addEdge('node3', 'node4')
+    .addEdge('node4', END);
+};
+
+test('basic agent execution', async () => {
+  const uncompiledGraph = createGraph();
+  const checkpointer = new MemorySaver();
+  const compiledGraph = uncompiledGraph.compile({ checkpointer });
+  const result = await compiledGraph.invoke(
+    { my_key: 'initial_value' },
+    { configurable: { thread_id: '1' } }
+  );
+  expect(result.my_key).toBe('hello from node4');
+});
+```
+:::
+
+## Testing individual nodes and edges
```

Comment:
On the python side, individual nodes will be standalone functions most of the time, which also makes them easy to test as one can test the function itself.

Perhaps we reserve graph.nodes as the more advanced version? (i.e., i personally wouldn't test my code with graph.nodes by default)